packages/console/src/containers/AppContent/components/Topbar/TenantSelector/index.tsx

```diff
@@ -102,42 +89,34 @@ function TenantSelector() {
           ))}
         </OverlayScrollbar>
         <Divider />
-        <div
-          role="button"
+        <button
           tabIndex={0}
           className={classNames(
             isCreateButtonDisabled && styles.disabled,
             styles.createTenantButton
           )}
+          disabled={isCreateButtonDisabled}
           onClick={() => {
-            if (isCreateButtonDisabled) {
-              return;
-            }
             setShowCreateTenantModal(true);
           }}
           onKeyDown={onKeyDownHandler(() => {
-            if (isCreateButtonDisabled) {
-              return;
-            }
             setShowCreateTenantModal(true);
           })}
         >
           <div>{t('cloud.tenant.create_tenant')}</div>
           <PlusSign />
-        </div>
+        </button>
       </Dropdown>
       <CreateTenantModal
         isOpen={showCreateTenantModal}
         onClose={async (tenant?: TenantInfo) => {
           if (tenant) {
-            void mutate();
-            window.location.assign(new URL(`/${tenant.id}`, window.location.origin).toString());
+            appendTenant(tenant);
```

Comment:
GET /tenants API response returns the tenant info array in the descending order of `createdAt` timestamp, in other words, the most recently created tenant should be placed at the beginning of the tenant array. Maybe we should use `prepend` instead of `append`.

Reply:
this is actually a good call but i doubt if we can maintain the sync of the ordering strategy once the server side changes.